release-notes/7.0/preview/api-diff/preview2/Microsoft.NETCore.App/7.0-preview2_System.Security.Cryptography.md

```diff
@@ -0,0 +1,220 @@
+# System.Security.Cryptography
+
+``` diff
+ namespace System.Security.Cryptography {
+     public sealed class AesCng : Aes {
++        protected override bool TryDecryptCbcCore(ReadOnlySpan<byte> ciphertext, ReadOnlySpan<byte> iv, Span<byte> destination, PaddingMode paddingMode, out int bytesWritten);
```

Comment:
Were these diffs done from the ref assemblies, or implementation assemblies?

Most of the crypto changes here look like it's diffing the windows (old) to Linux (new) implementation assemblies (or the ref assemblies against the Linux implementation assemblies).

None of the override changes in this file are substantial, and I'd greatly prefer if they were removed.

Only the static HashData and HashDataAsync methods in this file are representing real changes.

Reply:
I generated the diffs using the DLLs contained in the nupkgs that I downloaded from:
https://www.nuget.org/packages/Microsoft.AspNetCore.App.Ref/
https://www.nuget.org/packages/Microsoft.WindowsDesktop.App.Ref/
https://www.nuget.org/packages/Microsoft.NETCore.App.Ref/

And used the files inside the folder `/ref/net7.0/`.

Reply:
Strange.  Maybe it's just an artifact of the libraries getting squished together.

Reply:
Let me try to remove them. The AsmDiff tool has a CLI argument that allows showing "base member overrides" + "explicit interface implementations" together. I modified the code to see if I can split those two arguments into two, and I will only use the "explicit interface implementations" one.

Reply:
Nope, can't remove them. They are definitely showing up even if I remove the arguments.